diff --git a/node_modules/@web3-onboard/walletconnect/dist/walletConnect.js b/node_modules/@web3-onboard/walletconnect/dist/walletConnect.js
index bc94049..2714c0e 100644
--- a/node_modules/@web3-onboard/walletconnect/dist/walletConnect.js
+++ b/node_modules/@web3-onboard/walletconnect/dist/walletConnect.js
@@ -117,7 +117,9 @@ function walletConnect(options) {
                                 const hexChainId = isHexString(chainId)
                                     ? chainId
                                     : `0x${chainId.toString(16)}`;
-                                this.emit('chainChanged', hexChainId);
+                                setTimeout(() => {
+                                    this.emit('chainChanged', hexChainId);
+                                }, 1);
                             },
                             error: console.warn
                         });
@@ -235,6 +237,15 @@ function walletConnect(options) {
                                     ]
                                 });
                             }
+                            if (method === 'wallet_getPermissions') {
+                                return [{
+                                    parentCapability: 'eth_accounts',
+                                    caveats: [{
+                                    type: 'restrictReturnedAccounts',
+                                    value: this.connector.accounts,
+                                    }],
+                                }];
+                            }
                             return this.connector.request({
                                 method,
                                 params
